packages/Parallel/Application/ParallelFileProcessor.php

```diff
@@ -102,14 +104,30 @@ public function process(
         };
 
         $timeoutInSeconds = $this->parameterProvider->provideIntParameter(Option::PARALLEL_JOB_TIMEOUT_IN_SECONDS);
+        $useSeparateProcessPerJob = $this->parameterProvider->provideBoolParameter(Option::PARALLEL_USE_SEPARATE_PROCESS_PER_JOB);
+
+        if ($useSeparateProcessPerJob) {
+            $jobByProcessIdentifier = [];
+
+            foreach ($jobs as $job) {
+                $parallelProcessWithIdentifier = $this->createParallelProcessWithIdentifier($mainScript, $input, $serverPort, $streamSelectLoop, $timeoutInSeconds);
+                $processIdentifier = $parallelProcessWithIdentifier->getIdentifier();
+
+                $jobByProcessIdentifier[$processIdentifier] = $job;
+            }
+
+            $jobs = $jobByProcessIdentifier;
+            $jobIdentifiers = array_keys($jobByProcessIdentifier);
```

Comment:
when use separate process per job is enabled, the `$jobs` will become persistent with each job having as index the process identifiers.
the new `$jobIdentifiers` will be the new stack from which each process will pop the next one to run on exit